File: AGENT_2_ENGINE/engine/node_c_ppt.py
from pptx import Presentation
import os

class PPTGenerator:
    def __init__(self, manifest):
        self.manifest = manifest
        
    def generate(self, template_name, content_map, output_name):
        template_path = os.path.join(self.manifest['paths']['templates'], template_name)
        prs = Presentation(template_path)
        
        # Logic: Always clone from Slide Layouts (Indices) to preserve Master backgrounds
        # We assume index 1 is a standard content layout for this example
        layout = prs.slide_layouts[1] 
        slide = prs.slides.add_slide(layout)
        
        for shape in slide.placeholders:
            # We map content based on placeholder name or index
            if shape.name in content_map:
                shape.text = content_map[shape.name]
                
        output_path = os.path.join(self.manifest['paths']['outputs'], f"{output_name}.pptx")
        prs.save(output_path)
        return output_path

File: SOVEREIGN_CORE_V2/engine/engine/node_c_ppt.js
const PptxGenJS = require("pptxgenjs");
const path = require("path");

class PPTGenerator {
    constructor(manifest) {
        this.manifest = manifest;
    }

    async generate(templateName, contentMap, outputName) {
        let pptx = new PptxGenJS();
        
        // Note: pptxgenjs handles templates differently than python-pptx.
        // For this implementation, we simulate the 'Master' by defining a layout.
        pptx.layout = "LAYOUT_4x3";

        let slide = pptx.addSlide();
        
        // Title
        slide.addText(contentMap["Title 1"] || "Sales Presentation", {
            x: 0.5, y: 0.5, w: "90%", h: 1,
            fontSize: 32, color: "363636", bold: true
        });

        // Body Content (The Concise Persona Rewrite)
        slide.addText(contentMap["Text Placeholder 2"] || "", {
            x: 0.5, y: 1.5, w: "90%", h: 3,
            fontSize: 18, color: "666666", valign: "top"
        });

        // Background simulation (The 'Vibe' requirement: preserve backgrounds)
        // In a full implementation, we would use pptx.defineLayout with background images.

        const outputPath = path.join(this.manifest.paths.outputs, `${outputName}.pptx`);
        await pptx.writeFile({ fileName: outputPath });
        return outputPath;
    }
}

module.exports = PPTGenerator;
